File: scripts/card.py
class Card:
  def __init__(self, name, status, rarity, version, type, price) -> None:
    self.name = name
    self.status = status
    self.rarity = rarity
    self.version = version
    self.type = type
    self.price = price

  def __str__(self) -> str:
    return f'name: {self.name}, status: {self.status}, rarity: {self.rarity}, version: {self.version}, type: {self.type}, price: {self.price}'

  def to_sql_object(self):
    if self.status == '' or self.status == '仕様':
      return None
    return [self.name, self.status, self.rarity, self.version, self.type, self.price]

File: app/entity/card.go
package entity

import "time"

type Card struct {
	Name      string    `db:"name" json:"name"`
	Version   string    `db:"version" json:"version"`
	Rarity    string    `db:"rarity" json:"rarity"`
	Type      string    `db:"type" json:"type"`
	Price     int       `db:"price" json:"price"`
	Status    string    `db:"status" json:"status"`
	FetchDate time.Time `db:"fetch_date"`
}

func NewCard(
	name string,
	version string,
	rarity string,
	_type string,
	price int,
	status string,
	fetch_date time.Time,
) *Card {
	return &Card{
		Name:      name,
		Version:   version,
		Rarity:    rarity,
		Type:      _type,
		Price:     price,
		Status:    status,
		FetchDate: fetch_date,
	}
}
